runtime/browser/xwalk_browser_context.cc

```diff
@@ -109,11 +109,13 @@ XWalkBrowserContext::CreateZoomLevelDelegate(
 base::FilePath XWalkBrowserContext::GetPath() const {
   base::FilePath result;
 #if defined(OS_ANDROID)
-  CHECK(PathService::Get(base::DIR_ANDROID_APP_DATA, &result));
   base::CommandLine* cmd_line = base::CommandLine::ForCurrentProcess();
-  if (cmd_line->HasSwitch(switches::kXWalkProfileName))
-    result = result.Append(
-        cmd_line->GetSwitchValuePath(switches::kXWalkProfileName));
+  if (cmd_line->HasSwitch(switches::kUserDataDir))
+    result = cmd_line->GetSwitchValuePath(switches::kUserDataDir);
+  if (result.empty())
+    CHECK(PathService::Get(base::DIR_ANDROID_APP_DATA, &result));
+  result = result.Append(
```

Comment:
For android, profile name was set as default. This check can be ignored, but the check will increase the robustness, I will add it.